### PR TITLE
littlefs : Add LFS_READONLY and CUSTOM_LFS_CONFIG_FILE to Kconfig

### PR DESCRIPTION
**Pull request by poojamg**
Add littlefs config file and add lfs readonly flags to kconfig

Signed-off-by: Pooja Ganesh <poojamg@amperecomputing.com>

### CODE REVIEW

**Comment by poojamg** [comment]
@dleach02, Please can you help with review? Thanks!

File: subsys/fs/Kconfig.littlefs [hunk 1]
```diff
@@ -104,4 +104,24 @@ config FS_LITTLEFS_HEAP_PER_ALLOC_OVERHEAD_SIZE
...
+	help
+	  Enable lfs custom config file. This flag will allow CUSTOM_LFS_CONFIG_FILE.
+
+config CUSTOM_LFS_CONFIG_FILE
+	string "Custom LFS configuration file" if LFS_CFG
+	default "littlefs_util.h"
+	help
+	  Allow user defined input file for the LFS_CONFIG setting. The default config
+	  file can be tweaked with Kconfig to optimize resources with lfs library.
+	  You can specify the actual configuration file using the LFS_CONFIG.
```
Comment (tejlmand) [inline subsys/fs/Kconfig.littlefs:125]: we already have Kconfig for configuring LittleFS, so why do we need a second way of configuring LittleFS.

If users also have to consider a little fs header file for configuration then which configuration takes precedence Kconfig / header file.

How are the settings in the config header documented ?

When including files through Kconfig, then what will be the include path used ?
And what about absolute paths ?

Are there any settings in the config header that has dependencies to other settings ?
Kconfig allows dependencies to be described.

**Review by tejlmand — CHANGES_REQUESTED** [review]
Thanks for this draft, but I don't believe offering a second, and in Zephyr context undocumented, way of providing another configuration scheme for LittleFS is the right approach.

We already have Kconfig for configuration and if all modules starts to introduce additional config headers then this will grow into a big mess in future.

Need a very good reason to support this direction.

**Comment by poojamg** [comment]
Closing this PR, since we are going ahead with
https://github.com/zephyrproject-rtos/zephyr/issues/40559

**Comment by tejlmand** [comment]
@poojamg I assume you still want to introduce:
```

config LFS_READONLY
	bool "Enable readonly little fs"
	default 0
	help
	  Enable readonly littlefs
```
in the Kconfig tree, right ?
#40559 doesn't address that use-case.

**Comment by poojamg** [comment]
> @poojamg I assume you still want to introduce:
> 
> ```
> 
> config LFS_READONLY
> 	bool "Enable readonly little fs"
> 	default 0
> 	help
> 	  Enable readonly littlefs
> ```
> 
> in the Kconfig tree, right ? #40559 doesn't address that use-case.

yes, that is correct, I am waiting for your changes to be merged, I will need zephyr_lfs_config.template ..

**Pull request closed without merging**